### PR TITLE
Log backfill errors

### PR DESCRIPTION
**Pull request by joshkay10**
**What's in this PR?**
Extra logs to see why GitHub is failing the requests and check whether it keeps trying to push repository as the next task after failure

### CODE REVIEW

File: src/github/client/github-client-interceptors.ts [hunk 1]
```diff
@@ -138,6 +138,7 @@ export const handleFailedRequest = (logger: Logger) =>
 				logger.error(errorMessage);
 			}
 
+			logger.info({ response, request, error, isWarning }, "joshkay temp logging - githubclient interceptor handleFailedRequest");
```
Comment (joshkay10) [inline src/github/client/github-client-interceptors.ts:141]: Want the actual error on Axios fail

File: src/sync/installation.ts [hunk 2]
```diff
@@ -331,6 +331,7 @@ export const handleBackfillError = async (err,
 	logger: Logger,
 	scheduleNextTask: (delayMs: number) => void): Promise<void> => {
 
+	logger.info({ err, data, nextTask }, "joshkay temp logging - handleBackfillError");
```
Comment (joshkay10) [inline src/sync/installation.ts:334]: My current guess is that after it tries X times and finally calls the task a fail, its picking up "repository" as the next task again every time and can't get past it. This is here to try prove or disprove that nextTask is stuck on "repository"

File: src/github/client/github-client-interceptors.ts (hunk 1 above)
Comment (gxueatlassian) [reply src/github/client/github-client-interceptors.ts:141]: Is there a reason we use `info` instead of `error` for log level?

Comment (joshkay10) [reply src/github/client/github-client-interceptors.ts:141]: purely just was re-using the same method for all of my logs. This one is dumping out everthing it gets its hands on so not just specifically errors. Plus the hidden lines above are throwing .errrors

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**